.github/workflows/tests-main.yaml

```diff
@@ -24,9 +24,21 @@ jobs:
   unit:
     name: unit tests
     runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
     steps:
       - name: Checkout code
         uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - id: 'auth'
+        name: Authenticate to Google Cloud
+        uses: google-github-actions/auth@35b0e87d162680511bf346c299f71c9c5c379033 # v1.1.1
+        with:
+          workload_identity_provider: ${{ secrets.PROVIDER_NAME }}
+          service_account: ${{ secrets.SERVICE_ACCOUNT }}
+          access_token_lifetime: 600s
+          project_id: ${{ secrets.GOOGLE_CLOUD_PROJECT }}
+          create_credentials_file: true
```

Comment:
this line can be removed I believe, defaults to true

Reply:
Fixed.